cmd/cosign/cli/util.go

```diff
@@ -90,5 +90,6 @@ func DefaultRegistryClientOpts(ctx context.Context) []remote.Option {
 	return []remote.Option{
 		remote.WithAuthFromKeychain(authn.DefaultKeychain),
 		remote.WithContext(ctx),
+		remote.WithUserAgent("cosign/" + VersionInfo().GitVersion),
```

Comment:
maybe this isn't enough? any way to know if it's a release or non-release version?

Reply:
nvm, looks like `git describe --tags --always --dirty` does the right thing